### PR TITLE
Комшина Дарья. Задача 1. Вариант 4. Минимальное значение элемента в векторе.

### PR DESCRIPTION
**Pull request by MikasaAkc**
Алгоритм использует линейный поиск, который проходит по всем элементам массива и сравнивает каждый элемент с текущим минимальным значением.

Описание последовательной задачи.
Данная программа осуществляет поиск минимального элемента в заданном векторе целых чисел, применяя последовательный метод.

Описание MPI задачи
Данная программа распределяет задачу поиска минимального элемента в векторе целых чисел между несколькими процессами с использованием MPI. Основные этапы работы аналогичны последовательному подходу, но с использованием параллельных вычислений через MPI.

### CODE REVIEW

File: tasks/seq/komshina_d_min_of_vector_elements/func_tests/main.cpp
Comment (Dxppi) [inline tasks/seq/komshina_d_min_of_vector_elements/func_tests/main.cpp:1]: Add test with a zero vector

Comment (MikasaAkc) [reply tasks/seq/komshina_d_min_of_vector_elements/func_tests/main.cpp:1]: TEST(komshina_d_min_of_vector_elements_mpi, Empty_Vector) является тестом для проверки пустого(нулевого) вектора.

File: tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#include "mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <random>
+#include <string>
+#include <thread>
+#include <vector>
+
+using namespace std::chrono_literals;
```
Comment (allnes) [inline tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp:10]: please remove it

File: tasks/seq/komshina_d_min_of_vector_elements/src/ops_seq.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#include "seq/komshina_d_min_of_vector_elements/include/ops_seq.hpp"
+
+#include <thread>
+
+using namespace std::chrono_literals;
```
Comment (allnes) [inline tasks/seq/komshina_d_min_of_vector_elements/src/ops_seq.cpp:5]: please remove it

**Review by allnes — CHANGES_REQUESTED** [review]

File: tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp (hunk 1 above)
Comment (MikasaAkc) [reply tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp]: fixed

File: tasks/seq/komshina_d_min_of_vector_elements/src/ops_seq.cpp (hunk 2 above)
Comment (MikasaAkc) [reply tasks/seq/komshina_d_min_of_vector_elements/src/ops_seq.cpp:5]: fixed

Comment (MikasaAkc) [reply tasks/seq/komshina_d_min_of_vector_elements/src/ops_seq.cpp]: fixed

**Review by PutinVVV — APPROVED** [review]

File: tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+#include <boost/mpi/collectives.hpp>
+#include <boost/mpi/communicator.hpp>
+#include <memory>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "core/task/include/task.hpp"
```
Comment (NeeMiyuki) [inline tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp:14]: string and numeric are include but there is no sign that they are used and should be revomed. also remove a comment at the first line

**Review by NeeMiyuki — CHANGES_REQUESTED** [review]

File: tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp (hunk 3 above)
Comment (MikasaAkc) [reply tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp]: fixed)

**Review by allnes — APPROVED** [review]

**Review by NeeMiyuki — APPROVED** [review]

**Comment by aobolensk** [comment]
image

Please cover missing lines with tests

File: tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
+#include "mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <random>
+#include <string>
+
+int Min(const std::vector<int>& res) {
```
Comment (aobolensk) [inline tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp:8]: Consider using std::min_element

File: tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+#include <boost/mpi/communicator.hpp>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "core/task/include/task.hpp"
+
+namespace komshina_d_min_of_vector_elements_mpi {
+
+std::vector<int> getRandomVector(int sz);
```
Comment (aobolensk) [inline tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp:15]: Please, move test functions to the test module

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp (hunk 4 above)
Comment (MikasaAkc) [reply tasks/mpi/komshina_d_min_of_vector_elements/src/ops_mpi.cpp]: Fixed

File: tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp (hunk 5 above)
Comment (MikasaAkc) [reply tasks/mpi/komshina_d_min_of_vector_elements/include/ops_mpi.hpp]: Fixed(deleted because it is not used in tests)

**Review by aobolensk — APPROVED** [review]
